### PR TITLE
[python] add network config api

### PR DESCRIPTION
**Pull request by wxchan**
#985

### CODE REVIEW

File: python-package/lightgbm/basic.py [hunk 1]
```diff
@@ -1351,6 +1368,32 @@ def _free_buffer(self):
...
+        ----------
+        machines: list, set or string
+            List of machine names.
+        local_listen_port: int
+            TCP listen port for local machines.
+        listen_time_out: int
+            Socket time-out in minutes.
+        num_machines: int
+            The number of machines for parallel learning application
+        """
```
Comment (StrikerRUS) [inline python-package/lightgbm/basic.py]: Please update the docstring

```
        """Set the network configuration.

        Parameters
        ----------
        machines: list, set or string
            Names of machines.
        local_listen_port: int, optional (default=12400)
            TCP listen port for local machines.
        listen_time_out: int, optional (default=120)
            Socket time-out in minutes.
        num_machines: int, optional (default=1)
            The number of machines for parallel learning application.
        """
```

**Comment by guolinke** [comment]
@wxchan It that possible add some simulate test in travis ?

**Comment by wxchan** [comment]
@guolinke do you know how to simulate parallel learning on travis?

**Comment by guolinke** [comment]
@wxchan we it simulate it by multi-process ( in the same machine).  However, I am not sure about the in-bound port permission in Travis.

**Comment by guolinke** [comment]
@wxchan  close ?

**Comment by wxchan** [comment]
I forgot I used master branch for this pr and updated it by mistake. I will open a new one tonight.

发自我的 iPhone

在 2017年10月23日，上午9:04，Guolin Ke <notifications@github.com<mailto:notifications@github.com>> 写道：


@wxchan<https://github.com/wxchan> close ?

**Comment by wxchan** [comment]
move to #1019

**Pull request closed without merging**